js/gdm/authPrompt.js

```diff
@@ -600,18 +608,36 @@ const AuthPrompt = new Lang.Class({
         this.emit('cancelled');
     },
 
+    _loadCustomerSupportFile: function() {
+        if (this._customerSupportKeyFile)
+            return this._customerSupportKeyFile;
+
+        for (let path of CUSTOMER_SUPPORT_LOCATIONS) {
+            let keyFile = new GLib.KeyFile();
+            try {
+                keyFile.load_from_file(path, GLib.KeyFileFlags.NONE);
+                this._csutomerSupportKeyFile = keyFile;
+                return keyFile;
+            } catch (e) {
+                logError(e, 'Failed to read customer support data from ' + path);
+            }
+        }
+
+        return null;
+    },
+
     _ensureCustomerSupportData: function() {
         if (this._customerSupportEmail)
             return true;
 
-        const CUSTOMER_SUPPORT_FILENAME = 'vendor-customer-support.ini';
         const CUSTOMER_SUPPORT_GROUP_NAME = 'Customer Support';
         const CUSTOMER_SUPPORT_KEY_EMAIL = 'Email';
 
+        let keyFile = this._loadCustomerSupportFile();
+        if (!keyFile)
+            return false;
```

Comment:
Please `logError()` here as well

Reply:
There will be no error to log though... Since the above function already logs an error, I think we should simply return here if it fails.

Reply:
That was the original code and I suggested logging here, but you're right it's logging anyway in the function so it's probably redundant. @senko Sorry for the back and forth, I think @cosimoc is right and we should not log anything here.

Reply:
Changed per suggestion, thanks @cosimoc (hi, btw, long time no see! :-)